tests/test_mainpy.py

```diff
@@ -1,17 +1,26 @@
+import sys
 import asyncio
-
 import pytest
 
 import mainpy as mp
 
 
-def patch_module(module):
+def patch_module(monkeypatch, module):
+    # Patch the module name in the frame
+    frame = sys._getframe(1).f_back  # noqa
```

Comment:
Using `inspect.currentframe()` here should help with readability. Especially since the `import inspect` overhead isn't much of a issue, here.